lib/widgets/dialogs/invalid_subscription_dialog.dart

```diff
@@ -0,0 +1,225 @@
+                    place.copyWith(
+                      subscriptionId: newSubscriptionId,
+                      isExpired: false,
+                    ),
+                  ),
+            );
+      }
+      setState(() {
+        currentState = States.successfullyResubscribed;
+      });
+
+      LoadingScreen.instance().hide();
+    }
+
+    Widget body = _ReadyToResubscribe(onReSubscribe: onReSubscribe);
+
+    switch (currentState) {
+      case States.readyToResubscribe:
+        body = _ReadyToResubscribe(onReSubscribe: onReSubscribe);
+        break;
+      case States.error:
+        body = _FailedToResubscribe(
+          failedToResubscribe: failedToResubscribe!,
+          onReSubscribe: onReSubscribe,
+        );
+        break;
+      case States.successfullyResubscribed:
+        body = const _SuccessfullyResubscribed();
+        break;
+    }
+
+    return AlertDialog(
+      title: Text(localizations.invalid_subscription_dialog_headline),
+      content: body,
+      actions: <Widget>[
+        TextButton(
+          onPressed: () {
+            Navigator.of(context).pop();
+          },
+          child: Text(
+            localizations.main_dialog_close,
+            style: TextStyle(color: Theme.of(context).colorScheme.secondary),
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class _ReadyToResubscribe extends StatelessWidget {
+  const _ReadyToResubscribe({required this.onReSubscribe});
+
+  final VoidCallback onReSubscribe;
+
+  @override
+  Widget build(BuildContext context) {
+    var localizations = context.localizations;
+    ThemeData theme = Theme.of(context);
+
+    return Column(
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        Text(
+          localizations.invalid_subscription_dialog_text,
+        ),
+        const SizedBox(
+          height: 10,
+        ),
+        TextButton(
+          style: TextButton.styleFrom(
+            backgroundColor: theme.colorScheme.secondary,
+          ),
+          onPressed: onReSubscribe,
+          child: Text(
+            localizations.invalid_subscription_dialog_resubscribe_button,
+            style: TextStyle(
+              color: theme.colorScheme.onSecondary,
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class _FailedToResubscribe extends StatelessWidget {
+  const _FailedToResubscribe({
+    required this.failedToResubscribe,
+    required this.onReSubscribe,
+  });
+  final RegisterAreaError failedToResubscribe;
+  final VoidCallback onReSubscribe;
+
+  @override
+  Widget build(BuildContext context) {
+    var localizations = context.localizations;
+    ThemeData theme = Theme.of(context);
+
+    return Column(
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        Text(
+          localizations.invalid_subscription_dialog_error_body,
+          style: const TextStyle(fontSize: 18),
+        ),
+        Text("${localizations.invalid_subscription_dialog_error_status_code} "
```

Comment:
It's better to put the status code in the localization itself.

```dart
localizations.invalid_subscription_dialog_error_status_code(failedToResubscribe.statusCode.toString()),
```

You can do this by using the following in the arb file:

```json
{
    "invalid_subscription_dialog_error_status_code": "HTTP Status code: {status_code}",
    "@invalid_subscription_dialog_error_status_code": {
        "placeholders": {
            "status_code": {
                "type": "String"
            }
        }
    }
}
```

The generated code then accepts the string and puts it directly into the localization. This is especially useful for languages where variables need to be in different places of the sentence.